### PR TITLE
Get docker working for M1

### PR DESCRIPTION
**Pull request by Sybrand**
- Change Dockerfile base image - buster didn't have M1 compatible image  (also easer i.m.h.o. to work with Ubuntu for local development)
- Make a number of changes to get python to run as non root
- Updated some "docker compose" commands - "docker-compose" is now redundant, compose is build into latest versions of docker.
- Allowing a more recent version of python, we need to update to the latest version of python across our base anyway!
# Test Links:
[Percentile Calculator](https://wps-pr-2250.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-2250.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2250.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2250.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2250.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Fire Behaviour Advisory](https://wps-pr-2250.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-2250.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-2250.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,10 +22,10 @@ docker-web-server:
 	docker run -p 3000:3000 wps/web
 
 docker-shell-api:
-	# Shell into the dev container.
-	docker-compose run --rm api bash
+	# Shell into the running dev container.
+	docker compose exec api bash
```
Comment (conbrad) [inline Makefile:26]: This is not a command on my machine but the original `docker-compose` is

**Review by conbrad — CHANGES_REQUESTED** [review]

File: api/Dockerfile [hunk 2]
```diff
@@ -19,44 +29,50 @@ RUN apt-get update --fix-missing && apt-get -y install r-base
...
-RUN cd /tmp && \
-    poetry install --no-root
+COPY --chown=worker:worker poetry.lock pyproject.toml ./
+RUN python3 -m pip install cachecontrol
+
+RUN poetry install
+
+# We can't have this inside pyproject.toml because the gdal version differs from platform to platform.
+# To figure out what version of pygdal you need, run gdal-config
+RUN poetry run python -m pip install pygdal==3.4.1.10
```
Comment (conbrad) [inline api/Dockerfile:68]: Does the latest `pygdal` (3.5.1.11) work?
https://pypi.org/project/pygdal/3.5.1.11/

File: Makefile (hunk 1 above)
Comment (Sybrand) [reply Makefile:26]: `docker-compose` is an implementation of Docker Compose V1, and has been deprecated. If you don't have `docker compose` it means you're running an older version of docker, and you should upgrade. `docker-compose` is an additional dependency, so apart from everything else, it simplifies the development environment.

I don't think we should continue to use a deprecated version, and should transition to v2.
https://docs.docker.com/compose/#compose-v2-and-the-new-docker-compose-command
https://www.docker.com/blog/announcing-compose-v2-general-availability/

File: api/Dockerfile (hunk 2 above)
Comment (Sybrand) [reply api/Dockerfile:68]: No. Latest version that comes with Ubuntu 22.04 is 3.4.1

File: Makefile (hunk 1 above)
Comment (conbrad) [reply Makefile:26]: Huh weird. I'm running: `Docker version 20.10.16, build aa7e414fdc`

But I use the `limactl` wrapper which I guess just supports the old command.

**Review by conbrad — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (Sybrand) [reply Makefile:26]: Maybe your `limactl` needs an update? lima with nerdctl seems to support it - so it's strange that it wouldn't work with docker?

Comment (conbrad) [reply Makefile:26]: I needed to manually symlink the `docker compose` command as a plugin:

Compose is now a Docker plugin. For Docker to find this plugin, symlink it:    mkdir -p ~/.docker/cli-plugins    ln -sfn $(brew --prefix)/opt/docker-compose/bin/docker-compose ~/.docker/cli-plugins/docker-compose
--

https://formulae.brew.sh/formula/docker-compose

File: Makefile [hunk 3]
```diff
@@ -3,11 +3,11 @@ docker-build:
...
 	# Still having issues? try: docker system prune
-	docker-compose build
+	docker compose build
 
 docker-run:
 	# Run docker in dev mode.
-	docker-compose up
+	docker compose up
 
 docker-db:
```
Comment (conbrad) [inline Makefile:12]: There's an old `docker-compose` in this command

**Review by andrea-williams — APPROVED** [review]
After much pain, the dockerfile builds and runs on my machine 👍

File: Makefile (hunk 3 above)
Comment (Sybrand) [reply Makefile:12]: In that case - updating all the other references too.

**Pull request merged**